rts/Sim/Misc/Team.cpp

```diff
@@ -349,8 +349,15 @@ void CTeam::SlowUpdate()
 			if (team->isDead)
 				continue;
 
-			const float edif = std::max(0.0f, (team->resStorage.energy * 0.99f) - team->res.energy) * de;
-			const float mdif = std::max(0.0f, (team->resStorage.metal * 0.99f) - team->res.metal) * dm;
+			//due to precision errors mdif/edif sometimes can be slightly >= than res. If team has no metal income
+			//this causes units with zero fire resources requirements to unable to fire
+			//due to CTeam::HaveResources() failed check
```

Comment:
`this causes units with zero fire resources requirements to be unable to fire`
`when CTeam::HaveResources() is evaluated, thus clamp edif/mdif on both sides`

Reply:
ok